### ISSUE TITLE
php info screen returns 'Array to string conversion' error

### ISSUE THREAD

**Issue by rachelrine** [craftcms/cms#2843]
### Description

I'm only seeing this error on the client's server. My local env (homestead, vagrant) and dev server (forge, linode) don't seem to be having the same issue running the same code. Please let me know if I can provide more environment settings, I'm not totally sure what might be relevant here.
<img width="1739" alt="screen shot 2018-05-03 at 4 42 44 pm" src="https://user-images.githubusercontent.com/6887134/39602181-6c17a93e-4ef2-11e8-9d26-67acfcadecc0.png">

Additionally, if I print the php info in a file outside Craft, I can see everything as expected.
<img width="950" alt="screen shot 2018-05-03 at 4 52 01 pm" src="https://user-images.githubusercontent.com/6887134/39602180-6c0ca764-4ef2-11e8-9e6b-ecb28a5ced73.png">

### Steps to reproduce

1. Settings
2. Server Info
3. PHP Info

### Additional info

- Craft version: 2.6.3015
- PHP version: v7.0.30
- Database driver & version: mysql v14.14
- Plugins & versions:

**Comment by brandonkelly** [comment]
That error doesn't even make any sense for that line. Any ideas @angrybrad ?

**Comment by angrybrad** [comment]
Weird... any chance that's on a public box and we could get some CP and (s)FTP/SSH access to it?  If so, could you send that over to support@craftcms.com?

**Comment by rachelrine** [comment]
I will follow up with the client and get back to you guys soon. Thanks!

**Comment by john-henry** [comment]
Receiving this now on a client sites, locally and on their hosted sites. Was this ever resolved?

**Comment by angrybrad** [comment]
@john-henry it wasn't... any chance you could send some credentials to support@craftcms.com so we can look into it?

**Comment by john-henry** [comment]
@angrybrad I will certainly do so...after the weekend.

**Comment by angrybrad** [comment]
Got some credentials from @rachelrine, but honestly stumped by this... I can comment out every other line in this method [except for this one](https://github.com/craftcms/cms/blob/develop-v2/src/controllers/UtilsController.php#L58) and still get the same error. 

Even have it narrowed down to the `INFO_VARIABLES` part of phpinfo()... i.e. can reproduce with `phpinfo(INFO_VARIABLES);`, which `-1` includes.  That will output the $_SERVER and $ENV variables.  So at first, I thought this was a PHP bug using the specific PHP version on that box (7.0.23).

But if I try to reproduce it in a standalone script on the same server, it works fine.  Even tried bootstrapping Craft in the script and calling phpinfo() output after with no luck.

You running the same PHP version, @john-henry?  If so, can either of you try updating to PHP 7.1 so we can at least see if it's version specific?

**Comment by john-henry** [comment]
Local site works fine and that is 7.1.12

Live site (Arcustech) is  7.0.30 and I see phpinfo fine on standalone script but when visited in control panel same error.

Switched to 7.0.27 and that works locally. Interestingly MAMP doesn't provide a 7.0.30 to download and skips it completely. 

I suppose upgrading to 7.1 isn't a bad idea so will try and schedule that in

**Comment by angrybrad** [comment]
To muddy the waters even more, we've got an Arcustech site running the same PHP 7.0.30 and I can't reproduce it there, either.

**Comment by john-henry** [comment]
Ok some environment info I didn't add originally. We are using Redis and Cloudlfare. The latter is escaping the admin with a page rule. You using Redis @rachelrine ?

**Comment by rachelrine** [comment]
No Redis, and updating to 7.1 didn't resolve it either.

**Comment by angrybrad** [comment]
Turns out this is an issue with the now abandoned [Backup Pro](https://mithra62.com/) plugin.  More precisely, one of its own Composer dependencies is doing something that makes a call to `phpinfo()` choke later in the request.

And it must be that plugin + a combination of something else environmental, because I still can't reproduce it locally, but thanks to @rachelrine's debugging help disabling that plugin does fix it on her box.

I tried glancing through the plugin's dependencies to see if I can easily spot some something odd that might be causing it, but it's a bit of a beast (lots of dependencies), so didn't make it far.

At this point, I'd recommend just not using that plugin as it is no longer being actively developed.
